handlers/auth.go

```diff
@@ -0,0 +1,112 @@
+package handlers
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/Daniorocket/cookit-backend/lib"
+	"github.com/Daniorocket/cookit-backend/models"
+	uuid "github.com/satori/go.uuid"
+	"golang.org/x/crypto/bcrypt"
+)
+
+func (d *Handler) Login(w http.ResponseWriter, r *http.Request) {
+	login := &models.Login{}
+	err := json.NewDecoder(r.Body).Decode(login)
+	if err != nil {
+		log.Println("Error:", err)
+		if err := CreateApiResponse(w, nil, http.StatusBadRequest, "failed", "Failed to decode json"); err != nil {
+			log.Println("Error create Api response:", err)
+		}
+		return
+	}
+	passDB, err := models.GetPasswordByUsernameOrEmail(d.Client, d.DatabaseName, login.Username)
+	if err != nil {
+		log.Println("Error:", err)
+		if err := CreateApiResponse(w, nil, http.StatusBadRequest, "failed", "Invalid username or password"); err != nil {
+			log.Println("Error create Api response:", err)
+		} //User doesn't exists
+		return
+	}
+	if err := bcrypt.CompareHashAndPassword([]byte(passDB), []byte(login.Password)); err != nil {
+		log.Println("Error:", err)
+		if err := CreateApiResponse(w, nil, http.StatusBadRequest, "failed", "Invalid username or password"); err != nil {
+			log.Println("Error create Api response:", err)
+		}
+		return
+	}
+	expirationTime := time.Now().Add(15 * time.Minute)
+	tokenString, err := lib.CreateJWT(login.Username, expirationTime)
+	if err != nil {
+		CreateApiResponse(w, nil, http.StatusBadRequest, "failed", "Failed to create JWT")
+		return
+	}
+	CreateApiResponse(w, lib.Token{
+		Token:          tokenString,
+		ExpirationTime: expirationTime.UTC().String(),
+	}, http.StatusOK, "success", "none")
```

Comment:
Kwestia kto co woli, ale ja lubię rozdzielać fragmenty kodu od siebie enterami co robią dany blok - wydaje mi się to bardziej czytelne np. tutaj by było

```go
zdekoduj body

pobierz hasło

porównaj hasło

przygotuj responsa i zwróć 
```

Reply:
Faktycznie to jest bardziej czytelne, ja zazwyczaj starałem się pisać w taki sposób, aby jak najmniej linijek miał kod, poprawiam zatem w wielu miejscach.